docs/bluetooth.md

```diff
@@ -0,0 +1,208 @@
+
+```python
+from homeassistant.components import bluetooth
+
+...
+
+entry.async_on_unload(
+    bluetooth.async_register_callback(
+        hass, _async_nespresso_found, {"local_name": "Prodigio_*")}, bluetooth.BluetoothScanningMode.ACTIVE
+    )
+)
+```
+
+The below example shows registering to get callbacks for a device with the address `44:33:11:22:33:22`.
+
+```python
+from homeassistant.components import bluetooth
+
+...
+
+entry.async_on_unload(
+    bluetooth.async_register_callback(
+        hass, _async_specific_device_found, {"address": "44:33:11:22:33:22")}, bluetooth.BluetoothScanningMode.ACTIVE
+    )
+)
+```
+
+### Fetch the shared BleakScanner instance
+
+Integrations that need an instance of a `BleakScanner` should call the `bluetooth.async_get_scanner` API. This API returns a wrapper around a single `BleakScanner` that allows integrations to share without overloading the system.
+
+```python
+from homeassistant.components import bluetooth
+  
+scanner = bluetooth.async_get_scanner(hass)
+```
+
+
+### Determine if there is scanner running
+
+The Bluetooth integration may be set up but has no connectable adapters or remotes. The `bluetooth.async_scanner_count` API can be used to determine if there is a scanner running that will be able to receive advertisements or generate `BLEDevice`s that can be used to connect to the device. An integration may want to raise a more helpful error during setup if there are no scanners that will generate connectable `BLEDevice` objects.
+
+```python
+from homeassistant.components import bluetooth
+
+count = bluetooth.async_scanner_count(hass, connectable=True)
+```
+
+### Subscribing to unavailable callbacks
+
+To get a callback when the Bluetooth stack can no longer see a device, call the `bluetooth.async_track_unavailable` API. For performance reasons, it may take up to five minutes to get a callback once the device is no longer seen.
+
+If the `connectable` argument is set to `True`, if any `connectable` controller can reach the device, the device will be considered available. If only non-connectable controllers can reach the device, the device will be considered unavailable. If the argument is set to `False`, the device will be considered available if any controller can see it.
+
+```python
+from homeassistant.components import bluetooth
+
+def _unavailable_callback(address: str) -> None:
+    _LOGGER.debug("%s is no longer seen", address)
+   
+cancel = bluetooth.async_track_unavailable(hass, _unavailable_callback, "44:44:33:11:23:42", connectable=True)
+```
+
+### Fetching the bleak `BLEDevice` from the `address`
+
+Integrations wishing to avoid the overhead of starting an additional scanner to resolve the address may call the `bluetooth.async_ble_device_from_address` API, which returns a `BLEDevice` if the `bluetooth` integration scanner has recently seen the device. Integration MUST fall back to connecting via the `address` if the `BLEDevice` is unavailable.
```

Comment:
The integration MUST

Reply:
https://github.com/home-assistant/developers.home-assistant/pull/1455